src/components/KTreeList/KTreeList.vue

```diff
@@ -0,0 +1,419 @@
+    if (parentItem) {
+      // set new parent
+      if (!parentItem.children) {
+        parentItem.children = []
+      }
+      parentItem.children.push(selectedItem)
+    }
+  }
+
+  // TS not figuring out originalParent may have been set
+  // @ts-ignore
+  if (originalParent) {
+    // remove the item from the original parent
+    if (!originalParent.children) {
+      originalParent.children = []
+    }
+    originalParent.children = originalParent.children.filter((child: TreeListItem) => child.id !== item.id)
+  }
+}
+
+const handleChangeEvent = () => {
+  if (props.parentId) {
+    emit('child-change', {
+      parent: props.parentId,
+      children: internalList.value,
+    })
+  } else {
+    emit('change', {
+      items: internalList.value,
+    })
+  }
+}
+
+const maxLevelReached = computed(() => {
+  return props.level > props.maxLevel
+})
+
+/**
+     * Recursive check to get the maximum depth of an object.
+     * Documented here: https://stackoverflow.com/a/48505969
+     */
+const getMaximumDepth = ({ children = [] }): number => {
+  return children.length === 0 ? 0 : 1 + Math.max(...children.map(getMaximumDepth))
+}
+
+const checkMove = (target: any) => {
+  const levelOfDropLocation = target.relatedContext?.component?.$attrs.level || 0
+  const itemToDrop = target.draggedContext?.element
+
+  // check the level of the deepest nested child in the item being dropped
+  let deepestLevel = 0
+  if (itemToDrop.children?.length) {
+    deepestLevel = getMaximumDepth(itemToDrop)
+  }
+
+  const sumLevel = levelOfDropLocation + deepestLevel
+  if (sumLevel > props.maxLevel) {
+    // disallow movement
+    return false
+  }
+
+  // allow movement
+  return true
+}
+
+const onStartDrag = (): void => {
+  dragging.value = true
+  setDragCursor(true)
+}
+
+const onStopDrag = (target: any): void => {
+  dragging.value = false
+  setDragCursor(false)
+
+  const item = target.draggedContext?.element
+  const parent = target.relatedContext?.element
+  if (item && parent) {
+    handleSetParent(item, parent)
+  }
+  key.value++
+}
+
+// override cursor when dragging
+const setDragCursor = (value: boolean) => {
+  // must be on html element to keep style applied no matter where they drag
+  const html = document.getElementsByTagName('html').item(0)
+
+  if (html) {
+    html.classList.toggle('k-tree-list-grabbing', value)
+  }
+}
+
+// watch for programmatic changes to modelValue
+watch(value, (newVal, oldVal) => {
+  if (JSON.stringify(newVal) !== JSON.stringify(oldVal)) {
+    internalList.value = newVal
+  }
+})
+
+watch(() => props.items, (newValue, oldValue) => {
+  // Only trigger the watcher if items actually change
+  if (JSON.stringify(newValue) === JSON.stringify(oldValue)) {
+    internalList.value = newValue
+  }
+})
+
+onMounted(() => {
+  if (props.items) {
+    internalList.value = props.items
+  } else if (props.modelValue) {
```

Comment:
question: Wouldn't `v-model` or `modelValue` take precedence over `props.items`?

Reply:
They should be doing one or the other, but sure, I can switch the ordering